### PR TITLE
src/main: delay startup of applications until event loop is ready

### PR DESCRIPTION
**Pull request by Consolatis**
This mostly allows for using scripts that trigger a labwc SIGHUP to work as expected. Before this patch, there was a race between the event loop starting up and the autostart scripts executing.

One example of such a use-case is dynamically setting the keyboard layout or cursor theme via environment variable based on feedback from a dbus service like discussed in #1588.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -177,24 +206,16 @@ main(int argc, char *argv[])
 
 	menu_init(&server);
```
Comment (domo141) [inline src/main.c:208]: instead of the change below, would 

(void)wl_event_loop_dispatch(server.wl_event_loop, 0);

work ?

Comment (Consolatis) [reply src/main.c:208]: Potentially that could work (although I didn't look into it in detail). However, I think we should not interfere with eventloop internals here.

**Comment by domo141** [comment]
right -- I see your point there.

**Comment by johanmalm** [comment]
Thanks :smile: Nice improvement.

**Pull request merged**